sqlglot/dialects/tsql.py

```diff
@@ -854,6 +854,7 @@ class Generator(generator.Generator):
             exp.AutoIncrementColumnConstraint: lambda *_: "IDENTITY",
             exp.DateAdd: date_delta_sql("DATEADD"),
             exp.DateDiff: date_delta_sql("DATEDIFF"),
+            exp.Count: rename_func("COUNT_BIG"),
```

Comment:
Should we adjust the parsing a bit so that we can actually preserve `COUNT` calls and only output `COUNT_BIG` if 1) T-SQL parsed a `COUNT_BIG` or 2) we're transpiling `COUNT` from other dialects to T-SQL?

Reply:
yep, we should preserve round trip

Reply:
So @colin-ho, the action items here are:

- Add new arg to `exp.Count`, something like `"bigint": False`
- Mimic what @VaggelisD did in this [commit](https://github.com/tobymao/sqlglot/commit/1284fd0a64890d3548af7ed0a0cc05bb6166ccb2#diff-63eb8dd82d3561bc414e3e13cf59516bd6584c5766a10072157a8a5aee4ad85fR225) for `Coalesce` (needed because of `is_var_len_args`)
- Include a couple of `FUNCTION` entries in T-SQL to parse `COUNT` and `COUNT_BIG` appropriately
- Test roundtrip & transpilation
- Bonus: fix `exp.Count` type inference by removing [this](https://github.com/tobymao/sqlglot/blob/main/sqlglot/dialects/dialect.py#L531) and adding an entry in `ANNOTATORS` that sets to `INT`/`BIGINT` accordingly. Make sure to test.